Use more efficient `SplitSeq` instead of `Split`

tklauser opened the pull request:
Replace `Split` in `for range strings.Split(...)` by Go 1.24's more efficient, iterators-based `strings.SplitSeq`.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes with some minor manual edits.

kaworu reviewed (approved):
Thanks!

smagnani96 reviewed (approved):
Changes for encryption files LGTM, thanks!

nathanjsweet reviewed (approved):
Cool!

tommyp1ckles reviewed (approved):
👍 on metrics

tklauser committed:
Use more efficient SplitSeq instead of Split

Replace Split in "for range strings.Split(...)" by Go 1.24's more
efficient, iterators-based strings.SplitSeq.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes with some minor manual edits.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

pchaigno reviewed (approved):
Reviewing for contributing because it makes me sad to see such a straightforward PR having to go through that many reviews.
